### PR TITLE
Updates to Tables spec

### PR DESCRIPTION
**Pull request by annatisch**
Addresses the comments from PR https://github.com/Azure/azure-rest-api-specs/pull/8151

### CODE REVIEW

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (christothes) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:607]: Should we hold off on implementing this until there is support for the `merge` verb in autorest? It seems that it is only truly supported as documented by storage.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (christothes) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: If not specified, an insert will be performed when no existing entity is found to update **and a replace will be performed if an existing entity is found.**"

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (christothes) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: a => A

**Review by christothes — COMMENTED** [review]
LGTM - Just the outstanding question on the merge operation.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:597]: I have to add this as a "PATCH" operation for now - as autorest does not support "MERGE" operations.
The PATCH operation should still work for a Storage endpoint.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:607]: Based on the meeting this morning - it seems that we should optimize for the Storage experience, so I'm in favour of leaving it in, even if Cosmos doesn't support it.

**Review by christothes — APPROVED** [review]

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed!

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed! And added to the same description for the PATCH.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (lmazuel) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Already method by default defined here

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (lmazuel) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Already the parameter name

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (lmazuel) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: I don't think this is a valid format, if you want a simple string just use no format

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (lmazuel) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: No format needed (even maybe invalid)

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (lmazuel) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: See before

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (lmazuel) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:229]: Need to override application/json;odata=minimalmetadata?

Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:229]: Yeah... maybe not essential as I don't know what the service does with the extra odata tag - however the documented header here is just "application/json"

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed!

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed!

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed!

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed!

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json]: Fixed!

**Comment by annatisch** [comment]
@shurd - this PR should be good to go if you have no additional comments/concerns.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (shurd) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:35]: I believe @lmazuel suggested in an email a while back we make this nometadata by default since metadata is odata only and not information on the resource. Did the thinking change here?

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (shurd) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:610]: Do we need to add this file as a part of this PR?

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (shurd) [inline specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:5]: Does this API version work for both Cosmos and Azure Tables? If yes, we can just remove the 2018 file Otherwise, these changes should be reflected there (they are duplicate files other than the reference to "version").

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:35]: In this case the metadata includes information on how the data should be deserialized along with Etag information on the entity, so I would prefer to include it by default.
Customers can then opt-out if they need high-perf.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:5]: I have been using this API version for both - which generally seems to be working.
Even if that's not the case, we can override the API version at the client, which I think is preferable to maintaining two identical specs. We should just remove the 2018 set if there are no other differences.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (annatisch) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:610]: Good catch - added. Also updated the Delete example with the added required parameter.

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (shurd) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:5]: Sure makes sense

File: specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json
Comment (shurd) [reply specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2019-02-02/table.json:35]: ok thanks for confirming

**Pull request merged**
